iothub_client/tests/blob_ut/blob_ut.c

```diff
@@ -343,7 +343,7 @@ TEST_FUNCTION(Blob_UploadMultipleBlocksFromSasUri_with_NULL_SasUri_fails)
     unsigned int httpResponse = HTTP_OK;
 
     ///act
-    BLOB_RESULT result = Blob_UploadMultipleBlocksFromSasUri(NULL, FileUpload_GetData_Callback, &context, &httpResponse, testValidBufferHandle, NULL, NULL, NULL);
+    BLOB_RESULT result = Blob_UploadMultipleBlocksFromSasUri(NULL, FileUpload_GetData_Callback, &context, &httpResponse, testValidBufferHandle, NULL, NULL, NULL, 0);
```

Comment:
Please add a test where we're passing a non-zero timeout here and make sure that the HTTPAPI_SetOption() is being executed.  See the blocks in `Blob_UploadMultipleBlocksFromSasUri_various_sizes_happy_path_Impl` like say below to give a flavor of how this works with optional certs.

```c
        if (certificate != NULL)
        {
            STRICT_EXPECTED_CALL(HTTPAPIEX_SetOption(IGNORED_PTR_ARG, "TrustedCerts", IGNORED_PTR_ARG));
        }
```

Reply:
Added!